mythril/laser/ethereum/state/constraints.py

```diff
@@ -1,77 +1,98 @@
 """This module contains the class used to represent state-change constraints in
 the call graph."""
 
+from mythril.laser.smt import Solver, Bool
+
+from typing import List, Optional
+from z3 import unsat
+
 
 class Constraints(list):
     """This class should maintain a solver and it's constraints, This class
     tries to make the Constraints() object as a simple list of constraints with
     some background processing.
 
-    TODO: add the solver to this class after callback refactor
     """
 
-    def __init__(self, constraint_list=None, solver=None, possibility=None):
+    def __init__(
+        self,
+        constraint_list: Optional[List[Bool]] = None,
+        is_possible: Optional[bool] = None,
+    ) -> None:
         """
 
-        :param constraint_list:
-        :param solver:
-        :param possibility:
+        :param constraint_list: List of constraints
+        :param is_possible: Whether it is possible to satisfy the constraints or not
         """
         super(Constraints, self).__init__(constraint_list or [])
-        self.solver = solver
-        self.__possibility = possibility
+        self.__default_timeout = 10
+        self.__is_possible = is_possible
 
-    def check_possibility(self):
+    def check_possibility(self) -> bool:
         """
-
-        :return:
+        :return: True/False based on the existence of solution of constraints
         """
-        return True
+        if self.__is_possible is not None:
+            return self.__is_possible
+        solver = Solver()
+        solver.set_timeout(self.__default_timeout)
+        for constraint in self[:]:
+            solver.add(constraint)
+        self.__is_possible = solver.check() != unsat
```

Comment:
```suggestion
        self.__is_possible = solver.check() == sat
```
Unknown should also be categorized as not possible I think

Reply:
for low timeouts such as 10ms there might be a case where it might not be solvable right, but solvable at the time of 100s.

Reply:
But As the constraints seems to increase lot of `unknown` answers seems to popup due to the timeout(after increasing timeout to 100ms) so we might just miss those paths right?

Reply:
Maybe we need to put a higher timeout there

Reply:
increasing it to 100ms itself slows mythril by 2X , so should we increase the timeout?